test/api/test_api.py

```diff
@@ -71,7 +72,10 @@ def test_ci_measurement_add():
     data = DB().fetch_one(query, (measurement.run_id, ), row_factory=psycopg.rows.dict_row)
     assert data is not None
     for key in measurement.model_dump().keys():
-        assert data[key] == measurement.model_dump()[key], Tests.assertion_info(f"{key}: {data[key]}", measurement.model_dump()[key])
+        if key == 'workflow':
+            assert data['workflow_id'] == measurement.model_dump()[key], Tests.assertion_info(f"workflow_id: {data['workflow_id']}", measurement.model_dump()[key])
+        else:
+            assert data[key] == measurement.model_dump()[key], Tests.assertion_info(f"{key}: {data[key]}", measurement.model_dump()[key])
 
 
 def todo_test_get_runs():
```

Comment:
Althoug unrelated to the PR: Why is there a todo function here?

Reply:
this is from the front-end refactor, when we changed "projects" into "runs"

previously this test was "get_projects", and used to be a simple call as a quick effective smoketest for the api. it inserted a  project directly into the DB called the endpoint to see if we got the same back.

after the refactor the concept of projects changed into timeline_projects, whose endpoint relied on a more complicated logic that also relied on the machines table, and this test felt like it needed a more extensive re-write.

At the same time there was also a lot of other work which was needed to get the tests back up running generally, which I prioritized first and ended up being time consuming and over time-budget. As we had other tests in the same file that served the purpose of smoketesting the API, this remained as a todo.

Looking at this with fresh eyes though its not hard to get a working test for either the timeline_projects or runs endpoint, and our API  is lacking good test coverage, so its worth doing, but outside the scope of this PR